File: CommonIn3SortedArrays/CommonIn3SortedArrays.cpp
/*
https://www.geeksforgeeks.org/problems/common-elements1132/1
*/

class Solution {
  public:
    vector<int> commonElements(vector<int> &a, vector<int> &b, vector<int> &c) {
        map<int, int>mpa, mpb;
        for(auto &vals: a){
            mpa[vals]++;
        }
        for(auto &vals: b){
            mpb[vals]++;
        }
        vector<int>res;
        set<int>unqst;
        for(auto &vals: c){
            unqst.insert(vals);
        }
        for(auto &stval: unqst){
            auto itr1=mpa.find(stval);
            auto itr2=mpb.find(stval);
            if(itr1!=mpa.end()&&itr2!=mpb.end()){
                res.push_back(stval);
            }
        }
        sort(res.begin(), res.end());
        return res;
    }
};

File: CommonIn3SortedArrays/CommonIn3SortedArrays.java
/*
https://www.geeksforgeeks.org/problems/common-elements1132/1
*/

class Solution {
    public ArrayList<Integer> commonElements(int[] a, int[] b, int[] c) {
        HashMap<Integer, Integer>mapForA=new HashMap<>();
        HashMap<Integer, Integer>mapForB=new HashMap<>();
        for(int val: a){
            if(mapForA.containsKey(val)){
                int currentCount=mapForA.get(val);
                mapForA.put(val, currentCount+1);
            } else{
                mapForA.put(val, 1);
            }
        }
        for(int val: b){
            if(mapForB.containsKey(val)){
                int currentCount=mapForB.get(val);
                mapForB.put(val, currentCount+1);
            } else{
                mapForB.put(val, 1);
            }
        }
        HashSet<Integer>uniqueVals=new HashSet<>();
        for(int val: c){
            uniqueVals.add(val);
        }
        ArrayList<Integer>resultantList=new ArrayList<>();
        Iterator itr=uniqueVals.iterator();
        while(itr.hasNext()){
            int currentVal=(int)itr.next();
            boolean containsInA=mapForA.containsKey(currentVal);
            boolean containsInB=mapForB.containsKey(currentVal);
            if(containsInA!=false&&containsInB!=false){
                resultantList.add(currentVal);
            }
        }
        Collections.sort(resultantList);
        return resultantList;
    }
}